### DISCUSSION TITLE
How can I get the size of a block and the width of a Mtext?

### DISCUSSION THREAD

**Discussion by LavenderGu** [mozman/ezdxf#277] [Q&A]
Hi mozman,

I have three questions when using ezdxf. 

1. How can I get the size of an existing block?
2. How can I get the width of a Mtext object?
3. How can I add a table in CAD by row and column?

Thank you! And  it's really a useful package!!!

Best regards,
Gu

**Comment by mbway** [comment]
I assume that by size you mean the bounding box of the entity in CAD coordinates?

getting the size of arbitrary entities (which would be required for getting the bounding box of a block) is challenging and requires about as much work as actually drawing the entity. I would suggest that you use the matplotlib backend to draw (only) the entity that you want to measure then get the bounding box of the data present in the axes after drawing. If you want a tighter bounding box you can iterate over the artists in the axes and do whatever you like with them.

Getting the size of text requires knowledge of what font being used and the measurements of that font. Again the simplest way is to render the text and measure it afterwards. Instead if you just want to get the width of a line of text (not taking the placement of the text such as rotation and scaling) you can use `MatplotlibBackend.get_text_line_width()` if you don't want to use the whole rendering pipeline.

If you wanted a more performant approach you could implement your own rendering backend which simply keeps track of a bounding box containing any entities which have been drawn rather than actually drawing them. This would be more work but would be a nice clean approach.

I don't know how to add tables with ezdxf so I can't help there I'm afraid

**Reply by LavenderGu** [reply]
Thanks for your suggestions.  I'm going to try it.

**Comment by mozman** [comment]
ACAD_TABLE is not yet supported by ezdxf. Maybe support will be added in the future.
